### ISSUE TITLE
azurerm_data_protection_backup_vault - ZoneRedundancy option not accepted

### ISSUE THREAD

**Issue by marcelloformica** [hashicorp/terraform-provider-azurerm#24530]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.6

### AzureRM Provider Version

3.87.0

### Affected Resource(s)/Data Source(s)

azurerm_data_protection_backup_vault

### Terraform Configuration Files

```hcl
resource "azurerm_data_protection_backup_vault" "bvault" {
  provider = azurerm.lz
  name                          = var.bvault_name
  location                      = var.bvault_location
  resource_group_name           = var.bvault_rg_name
  redundancy                    = var.bvault_redundancy
  datastore_type                = "VaultStore"
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

azurerm_data_protection_backup_vault  should support also zone redundancy, besides GeoRedundant  and LocallyRedundant types, so that we have a feature parity with (for example), creation from Azure Portal.

In https://github.com/hashicorp/terraform-provider-azurerm/issues/21300  it was explained that this requires API version bump from 2022-04-01 to 2023-01-01; I see that this version upgrade has been done later, in https://github.com/hashicorp/terraform-provider-azurerm/pull/24143,   is anything else required to allow this redundancy type?

### Actual Behaviour

Error: expected redundancy to be one of ["GeoRedundant" "LocallyRedundant"], got ZoneRedundant

### Steps to Reproduce

terraform apply of the above sample code, passing 'ZoneRedundant' as value for var.bvault_redundancy

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/21300
https://github.com/hashicorp/terraform-provider-azurerm/pull/24143

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. I do see that the version was updated in #24143, which means that the original enhancement request in #21300 is no longer blocked by a version update. We like to try to keep discussions consolidated, so we’re going to close this new issue in favor of #21300 since the two issues are asking for the same enhancement.

**Comment by rcskosir** [comment]
*changed the closure type to match duplicate

**Comment by sinbai** [comment]
Hi @marcelloformica I have submitted PR #24556 to support `ZoneRedundancy ` for `azurerm_data_protection_backup_vault`. Could you please track it for more updates?
